src/utils/flux/connect/connect.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+
+/**
+ * Higher order component
+ *
+ * Takes a variadic number of arguments where the leading arguments are
+ * Stores and the the tail of those arguments is a function mapping from
+ * store states to the props of the composed component.
+ *
+ * Usage like so:
+ *
+ *    function mapStateToProps(stateA, stateB) {
+ *      return {
+ *        propFromA: stateA.get('someProp'),
+ *        propFromB: stateB.get('someProp')
+ *      }
+ *    }
+ *
+ *    const ConnectedComponent = connect(storeA, storeB, mapStateToProps)(TargetComponent)
+ *
+ * @function connect
+ * @param {...Store} ...stores
+ * @param {Function} mapStateToProps
+ * @return {Function}
+ */
+const connect = (...args) => (WrappedComponent) => {
+  const stores = args.slice(0, args.length - 1);
+  const mapToProps = args[args.length - 1];
+
+  class ConnectedComponent extends React.Component {
+    componentDidMount() {
+      this.unsubscribers = stores.map((store) => {
+        const updateComponent = () => { this.forceUpdate(); };
```

Comment:
Would there be any benefit to maintaining state rather than triggering `forceUpdate`? (eg. could help with debugging? could get some performance gains? would remove use of unrecommended function?)

Reply:
I'm not sure. I think as a first pass its worth sticking with `forceUpdate`, as I don't foresee it causing any major issues initially. There will certainly be some optimisation required in future, but probably best to try and tackle that organically if/when its required.

Reply:
OK - my main concern really was if we're losing any of the batching perfs etc that React does behind the scenes - but it [looks as though it treats it the same](https://github.com/facebook/react/blob/d9c1dbd61772f8f8ab0cdf389e70463d704c480b/packages/react-dom/src/__tests__/ReactUpdates-test.js#L256-L296) as `setState`.

Reply:
One other benefit of using state would be making the migration path easier. Looking at a lot of the tests currently in the application code they rely on setting state to manipulate the store data currently rendered. However I'm not entirely sure what storing the state would look like (would we store the entire store data in state or just the mapped props?), and I'm not sure what the tests would look like if we didn't use state (would they manipulate the data on the store instance and force a re-render?).

Reply:
I'm guessing it should be a simple case of referencing store state through the wrapped component's `props` rather than its `state`, for existing implementations. Components shouldn't be manipulating store data outside of the stores' subscribers.

I'm sure what you mean by the last question. Do you mean what would tests look like for `mapStateToProps` or what would the wrapped component's tests look like?

Reply:
Also, HOCs don't share component `state`, rather they pass `props` down the chain.